watertap/property_models/anaerobic_digestion/adm1_reactions.py

```diff
@@ -0,0 +1,1734 @@
+        """
+        Callable method for Block construction
+        """
+        super().build()
+
+        # Create references to state vars
+        # Concentration
+        add_object_reference(self, "conc_mass_comp_ref", self.state_ref.conc_mass_comp)
+
+        self.rates = {
+            "R1": 1.786e-06,
+            "R2": 3.235e-06,
+            "R3": 1.187e-05,
+            "R4": 3.412e-06,
+            "R5": 3.404e-06,
+            "R6": 1.187e-05,
+            "R7": 3.185e-06,
+            "R8": 2.505e-06,
+            "R9": 3.230e-06,
+            "R10": 2.636e-06,
+            "R11": 1.220e-05,
+            "R12": 4.184e-06,
+            "R13": 9.726e-08,
+            "R14": 2.730e-07,
+            "R15": 5.626e-08,
+            "R16": 9.998e-08,
+            "R17": 3.178e-08,
+            "R18": 1.761e-07,
+            "R19": 7.338e-08,
+        }
+
+    # Rate of reaction method
+    def _rxn_rate(self):
+        self.reaction_rate = pyo.Var(
+            self.params.rate_reaction_idx,
+            initialize=self.rates,
+            bounds=(1e-9, 1e-4),
+            doc="Rate of reaction",
+            units=pyo.units.kg / pyo.units.m**3 / pyo.units.s,
+        )
+
+        self.conc_mass_va = pyo.Var(
+            initialize=0.01159624,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of va-",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+
+        self.conc_mass_bu = pyo.Var(
+            initialize=0.0132208,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of bu-",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+        self.conc_mass_pro = pyo.Var(
+            initialize=0.015742,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of pro-",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+        self.conc_mass_ac = pyo.Var(
+            initialize=0.1972,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of ac-",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+        self.conc_mol_hco3 = pyo.Var(
+            initialize=0.142777,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of hco3",
+            units=pyo.units.kmol / pyo.units.m**3,
+        )
+        self.conc_mol_nh3 = pyo.Var(
+            initialize=0.004,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of nh3",
+            units=pyo.units.kmol / pyo.units.m**3,
+        )
+        self.conc_mol_co2 = pyo.Var(
+            initialize=0.0099,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of co2",
+            units=pyo.units.kmol / pyo.units.m**3,
+        )
+        self.conc_mol_nh4 = pyo.Var(
+            initialize=0.1261,
+            domain=pyo.NonNegativeReals,
+            doc="molar concentration of nh4",
+            units=pyo.units.kmol / pyo.units.m**3,
+        )
```

Comment:
Are these included in the rxn model instead of properties because these correspond to products of reactions as opposed to reactants?Tagging #909 to come back to this.

Reply:
Yes, and they are only used for the charge balance and the CO2 conversion.